core/common/src/main/java/alluxio/collections/IndexedSet.java

```diff
@@ -255,46 +231,40 @@ public void remove() {
   }
 
   /**
-   * Whether there is an object with the specified field value in the set.
+   * Whether there is an object with the specified unique index field value in the set.
    *
-   * @param index the field index
+   * @param indexDefinition the field index definition
    * @param value the field value
    * @return true if there is one such object, otherwise false
    */
-  public boolean contains(FieldIndex<T> index, Object value) {
-    ConcurrentHashSet<T> set = getByFieldInternal(index, value);
-    return set != null && !set.isEmpty();
+  public boolean contains(IndexDefinition<T> indexDefinition, Object value) {
+    FieldIndex<T> index = mIndices.get(indexDefinition);
+    return index != null && index.contains(value);
   }
 
   /**
    * Gets a subset of objects with the specified field value. If there is no object with the
-   * specified field value, a newly created empty set is returned. Otherwise, the returned set is
-   * backed up by an internal set, so changes in internal set will be reflected in returned set.
+   * specified field value, a newly created empty set is returned.
    *
-   * @param index the field index
+   * @param indexDefinition the field index definition
    * @param value the field value to be satisfied
    * @return the set of objects or an empty set if no such object exists
    */
-  // TODO(gpang): Remove this method, if it is not being used.
-  public Set<T> getByField(FieldIndex<T> index, Object value) {
-    Set<T> set = getByFieldInternal(index, value);
-    return set == null ? new HashSet<T>() : Collections.unmodifiableSet(set);
+  public Set<T> getByField(IndexDefinition<T> indexDefinition, Object value) {
+    FieldIndex<T> index = mIndices.get(indexDefinition);
+    return index == null ? Collections.<T>emptySet() : index.getByField(value);
```

Comment:
we should throw an exception if the index can't be found - this means that the caller is trying to use an index that they didn't define the IndexedSet to include.